src/peft/tuners/tuners_utils.py

```diff
@@ -363,6 +363,30 @@ def _prepare_model(self, peft_config: PeftConfig, model: nn.Module):
         """
         pass
 
+    @staticmethod
+    def _check_tied_module_exists(peft_config: PeftConfig, key: str) -> bool | re.Match[str] | None:
+        """
+        A helper method to check if the passed module's key name matches any of the tied modules
+
+        Args:
+            config (`PeftConfig`):
+                A config to match target modules from.
+            key (`str`):
+                A key to search any matches in config.
+
+        Returns:
+            `bool`
+                True if key matches any tied modules from config, False if no match found.
+        """
+        _target_modules_to_tie = getattr(peft_config, "target_modules_to_tie", {}) or {}
+
+        if key in _target_modules_to_tie or any(
+            key.endswith(f".{target_key}") for target_key in _target_modules_to_tie
+        ):
+            return True
+
+        return False
```

Comment:
```suggestion
        target_modules_to_tie = getattr(peft_config, "target_modules_to_tie", []) or []
        return key in target_modules_to_tie or any(
            key.endswith(f".{target_key}") for target_key in target_modules_to_tie
        )
```

Reason:
1. More concise
2. Doesn't really make sense to use dicts (`{}`) as default. If you meant to use the empty set, I don't think it amortizes here.